app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -0,0 +1,555 @@
+        .observeOn(AndroidSchedulers.mainThread())
+        .subscribe(new SingleObserver<String>() {
+          @Override public void onSubscribe(Disposable d) {
+          }
+
+          @Override public void onSuccess(String s) {
+            progressDialog.dismiss();
+            serviceIntent.putStringArrayListExtra(SELECTED_ZIM_PATHS_KEY, selectedBooksPath);
+            serviceIntent.setAction(ACTION_START_SERVER);
+            startService(serviceIntent);
+            Log.d(TAG, "onSuccess:  " + s);
+          }
+
+          @Override public void onError(Throwable e) {
+            // display the ip and don't forget to dismiss the dialog
+            progressDialog.dismiss();
+            Toast.makeText(ZimHostActivity.this, R.string.server_failed_message, Toast.LENGTH_SHORT)
+                .show();
+            Log.d(TAG, "Unable to turn on server", e);
+          }
+        });
+  }
+
+  void createHotspotIntent(String ACTION) {
+    serviceIntent.setAction(ACTION);
+    startService(serviceIntent);
+  }
+
+  void mobileDataDialog() {
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
+      AlertDialog.Builder builder = new AlertDialog.Builder(this, dialogStyle());
+
+      builder.setPositiveButton(this.getString(R.string.yes),
+          (dialog, id) -> openMobileDataActivity());
+      builder.setNegativeButton((android.R.string.no),
+          (dialog, id) -> startHotspotManuallyDialog());
+      builder.setTitle(this.getString(R.string.mobile_data_enabled));
+      builder.setMessage(
+          this.getString(R.string.mobile_data_message) + "\n" + this.getString(
+              R.string.mobile_data_message_confirmation)
+      );
+      builder.setCancelable(false);
+      AlertDialog dialog = builder.create();
+      dialog.show();
+    }
+  }
+
+  @Override public void onServerStarted(@NonNull String ipAddress) {
+    this.ip = ipAddress;
+    layoutServerStarted();
+  }
+
+  @Override public void onServerStopped() {
+    layoutServerStopped();
+    if (selectedBooksPath.size() > 0) {
+      selectedBooksPath.clear();
+    }
+  }
+
+  @Override public void onServerFailedToStart() {
+    Toast.makeText(this, R.string.server_failed_toast_message, Toast.LENGTH_LONG).show();
+  }
+
+  @Override public void onHotspotTurnedOn(@NonNull WifiConfiguration wifiConfiguration) {
+
+    hotspotService.startForegroundNotificationHelper();
+
+    //Show an alert dialog for hotspot details
+    AlertDialog.Builder builder = new AlertDialog.Builder(this, dialogStyle());
+
+    builder.setPositiveButton(android.R.string.ok, (dialog, id) -> {
+
+      progressDialog =
+          ProgressDialog.show(this, getString(R.string.progress_dialog_starting_server), "",
+              true);
+      progressDialog.show();
+
+      startFlowable();
+
+    });
+
+    builder.setTitle(this.getString(R.string.hotspot_turned_on));
+    builder.setMessage(
+        this.getString(R.string.hotspot_details_message) + "\n" + this.getString(
```

Comment:
should be a format string in strings.xml

Reply:
Fixed. e5bda9c395a25a1fa629de27f3553f86bea743bb